src/rpc/blockchain.cpp

```diff
@@ -1167,6 +1168,109 @@ static RPCHelpMan pruneblockchain()
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "File name and path for dump file are required");
+
+    fs::path filepath = fs::u8path(request.params[0].get_str());
+    filepath = fs::absolute(filepath);
+
+    if (fs::exists(filepath)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, filepath.u8string() + " already exists. If you are sure this is what you want, move it out of the way first");
+    }
+
+    fsbridge::ofstream file;
+    file.open(filepath);
+    if (!file.is_open())
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot open dump file");
+
+    if (!g_coin_stats_index) {
+        throw JSONRPCError(RPC_MISC_ERROR, "Coinstats index is not enabled");
+    }
+
+    file << strprintf("height,"
+                      "bestblock,"
+                      "txouts,"
+                      "bogosize,"
+                      "muhash,"
+                      "total_amount,"
+                      "total_unspendable_amount,"
+                      "total_prevout_spent_amount,"
+                      "total_new_outputs_ex_coinbase_amount,"
+                      "total_coinbase_amount,"
+                      "total_subsidy,"
+                      "total_genesis_block,"
+                      "total_bip30,"
+                      "total_unspendable_scripts,"
+                      "total_unclaimed_rewards\n");
+
+    CCoinsStats stats{CoinStatsHashType::MUHASH};
+    NodeContext& node = EnsureAnyNodeContext(request.context);
+    ChainstateManager& chainman = EnsureChainman(node);
+    CChainState& active_chainstate = chainman.ActiveChainstate();
+    active_chainstate.ForceFlushStateToDisk();
+    CBlockIndex *pindex = active_chainstate.m_chain.Genesis();
+
+    CCoinsView* coins_view;
+    BlockManager* blockman;
+    {
+        LOCK(::cs_main);
+        coins_view = &active_chainstate.CoinsDB();
+        blockman = &active_chainstate.m_blockman;
+    }
+
+    for (; pindex; pindex = active_chainstate.m_chain.Next(pindex)) {
+        if (!GetUTXOStats(coins_view, *blockman, stats, node.rpc_interruption_point, pindex))
+            throw JSONRPCError(RPC_INTERNAL_ERROR, "Unable to read UTXO set");
+
+        file << strprintf("%d,", static_cast<int64_t>(stats.nHeight));
+        file << strprintf("%s,", stats.hashBlock.GetHex());
+        file << strprintf("%d,", static_cast<int64_t>(stats.nTransactionOutputs));
+        file << strprintf("%d,", static_cast<int64_t>(stats.nBogoSize));
+        file << strprintf("%d,", stats.hashSerialized.GetHex());
+        CHECK_NONFATAL(stats.total_amount.has_value());
+        file << strprintf("%s,", FormatMoney(stats.total_amount.value()));
+        file << strprintf("%s,", FormatMoney(stats.total_unspendable_amount));
+        file << strprintf("%s,", FormatMoney(stats.total_prevout_spent_amount));
+        file << strprintf("%s,", FormatMoney(stats.total_new_outputs_ex_coinbase_amount));
+        file << strprintf("%s,", FormatMoney(stats.total_coinbase_amount));
```

Comment:
I would prefer these to be per-block amounts like in the RPC.

Reply:
There seemed to be a few people that preferred this version back when I first built the `coinstatsindex` but we can just have both. So now there is a bool argument to switch between cummulative and per-block.